framework/src/modules/chain/dpos/delegates_list.js

```diff
@@ -88,11 +88,30 @@ class DelegatesList extends EventEmitter {
 	}
 
 	async deleteDelegateListUntilRound(round) {
+		this.deleteDelegateListsFromCache(roundFromCache => roundFromCache < round);
 		await this.storage.entities.RoundDelegates.delete({
 			round_lt: round,
 		});
 	}
 
+	async deleteDelegateListAfterRound(round, tx) {
+		this.deleteDelegateListsFromCache(roundFromCache => roundFromCache > round);
+		await this.storage.entities.RoundDelegates.delete(
+			{
+				round_gt: round,
+			},
+			tx,
+		);
+	}
+
+	deleteDelegateListsFromCache(filterFn) {
+		Object.keys(this.delegateListCache)
```

Comment:
Probably makes more sense to do to more functionaly. Something like:
```
this.delegateListCache = Object.keys(this.delegateListCache).filter(filterFn)
    .reduce((newDelegateListCache, roundFromCacheToKeep) => { 
        newDelegateListCache[roundFromCacheToKeep] = this.delegateListCahe[roundFromCacheToKeep];
       return newDelegateListCache;
    }, {})

Reply:
But here, the filtered rounds have to be deleted. This function is keeping everything. Also, we'll get rid of cache mechanism #4217. So this code gonna be removed very very soon :D

Reply:
We are reassigning the variable: `this.delegateListCache` in the code I shared above. The value of this object will not contain the delegateList for rounds which fail the filter condition. So, it's irrelevant if we delete keys from the old delegateListCache object. Right? 
Also, this is not refactoring of some old code. I am suggesting changes in the code which was added as part of this PR. So, doesn't it make sense to add code which is cleaner?

Reply:
@SargeKhan filter conditions that were used in the other functions were returning only the rounds to be deleted. That's what i meant so reduce only iterates on the rounds to be deleted in that case. But we can negate those conditions. That's not a problem.

What I mean is, the whole caching logic will be deleted, so this function will be deleted completely. So there is no need to refactor it to be functional.